src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,26 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = do
+  b <- typeableBinder
+  indented *> doubleColon
+  t <- parseType
+  return $ TypedBinder t b
+    where typeableBinder = P.choice (map P.try
```

Comment:
Can you give an example where a binder would not be typeable? It's parsing most types except the top-level ones (as far as I understand), but when I tried to parse those as a part of this function, I ran into a space leak due to unbounded recursion I think. I'm still fuzzy on why it happened.